src/UICrafter.Core/Components/PocButtonNoMistakes.razor

```diff
@@ -0,0 +1,66 @@
+@using MudBlazor
+@using UICrafter.Core.Models
+@using System.Diagnostics
+@using Microsoft.AspNetCore.Components
+@using System.Net.Http
+@using System.Net.Http.Json
+@using UICrafter.Core.Utility
+
+@inject HttpClient Http
+
+
+<MudButton OnClick="CallApi"
+           Color="@ColorOptions.GetColor(Model)"
+           Disabled=@isDisabled>
+    @Model.Name
+</MudButton>
+
+@if (!string.IsNullOrEmpty(Model.JSONResponse))
+{
+    <MudPaper Elevation="3" Class="pa-4">
+        <MudText Typo="Typo.h4" GutterBottom="true">API Response:</MudText>
+        @Model.JSONResponse
+    </MudPaper>
+}
+
+
+@code {
+    private bool isDisabled = false;
+
+    [Parameter]
+    public PocButtonModel? Model { get; set; }
+
+    [CascadingParameter]
+    public StupidStore? StupidStore { get; set; }
+
+    private async Task CallApi()
+    {
+        try
+        {
+            isDisabled = true;
+            var tmp = Model.Name;
+            Model.Name = "Loading...";
+
+                var response = await Http.GetStringAsync(Model.URL);
+                
+                if (response == null)
```

Comment:
This is not how you do response checking. And generally it is bad form to use httpclient directly in component this should be abstracted into a handler or service.